airbyte-webapp/src/components/StreamTestingPanel/StreamTester.tsx

```diff
@@ -0,0 +1,122 @@
+import { faWarning } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { useEffect, useState } from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import { RotateIcon } from "components/icons/RotateIcon";
+import { Button } from "components/ui/Button";
+import { ResizablePanels } from "components/ui/ResizablePanels";
+import { Text } from "components/ui/Text";
+import { Tooltip } from "components/ui/Tooltip";
+
+import { StreamsListReadStreamsItem } from "core/request/ConnectorBuilderClient";
+import { useReadStream } from "services/connectorBuilder/ConnectorBuilderApiService";
+import { useConnectorBuilderState } from "services/connectorBuilder/ConnectorBuilderStateService";
+
+import { LogsDisplay } from "./LogsDisplay";
+import { ResultDisplay } from "./ResultDisplay";
+import styles from "./StreamTester.module.scss";
+
+interface StreamTesterProps {
+  selectedStream: StreamsListReadStreamsItem;
+}
+
+export const StreamTester: React.FC<StreamTesterProps> = ({ selectedStream }) => {
```

Comment:
Most of the code here is moved over from StreamTestingPanel and TestControls, so this isn't totally net-new. But it does contain some changes, mainly around handling errors from the `useReadStream` hook.

This component was created because I needed a single component containing the URL display, test button, and result display, so that the entire thing could be conditionally rendered only if there are streams available. If there are no streams detected, this entire component is not rendered, as desired.

**I'm open to naming suggestions on this component**, since it feels too close to StreamTestingPanel at the moment.